### ISSUE TITLE
AsyncWrap init hook threw

### ISSUE THREAD

**Issue by julien-f** [AndreasMadsen/trace#19]
Sometimes I get the error:

```
FATAL ERROR: node::AsyncWrap::AsyncWrap init hook threw
Aborted
```

Node v4.2.4 & trace v2.2.0.

**Comment by AndreasMadsen** [comment]
Try using node 5.3.0 there are some required internal features that node 4.2.4 doen't have. However I have been promised that they will land in version 4.2.5

**Comment by julien-f** [comment]
You should:

1. express this requirement in `package.json` in the `engines.node`
2. maybe even detect this at module startup and throw, because currently there is no way to handle this error and to fail gracefully (i.e. continue without trace).

**Comment by AndreasMadsen** [comment]
I'm not aware of a syntax that can set independent node version requirements for both v4 and v5. How does that look.

In any case the commits have already landed in v4-staging, so it is a very temporary issue: https://github.com/nodejs/node/pull/3461#issuecomment-167883042

**Comment by julien-f** [comment]
`node: '^4.2.5 || ^5.3'` should work.

What about code already using trace but before 4.2.5?
Currently the program aborts and it is not handled by [a try-catch](https://github.com/vatesfr/xo-server/blob/next-release/better-stacks.js#L9)…

**Comment by AndreasMadsen** [comment]
They should not upgrade to 2.2.x

**Comment by AndreasMadsen** [comment]
I guess semver was violated to that regard, I'm sorry about that.

**Comment by AndreasMadsen** [comment]
I have published:
* `2.2.1` with a better engines field that supports `^5.3`
* `2.0.2` with a better engines field that supports `^4.0`

`2.0.2` unfortunately don't have a fix for #17, that is not possible without the new features.

_Also, I don't plan to continue doing legacy updates. And hopefully it shouldn't be necessary when `4.2.5` gets released._
